packages/gatsby-plugin-mdx/src/plugin-options.ts

```diff
@@ -7,11 +7,10 @@ import rehypeMdxMetadataExtractor from "./rehype-metadata-extractor"
 import { remarkMdxHtmlPlugin } from "./remark-mdx-html-plugin"
 import { remarkPathPlugin } from "./remark-path-prefix-plugin"
 
-export interface IMdxPluginOptions extends PluginOptions {
+export interface IMdxPluginOptions extends Partial<PluginOptions> {
```

Comment:
Why partial?

Reply:
I think that was because of the required `plugin` property. --> https://github.com/gatsbyjs/gatsby/pull/35983#discussion_r909396741

Reply:
Not sure if you need to extend from PluginOptions. What benefit do you get from it?

Reply:
I did it to get TypeScript happy, but you are where right here, my approach was wrong.

Instead of extending them now, I process the ones given by Gatsby and turn them into IMdxPluginOptions.

https://github.com/gatsbyjs/gatsby/pull/35983/commits/21f2dfa8957fc823ea7576e8fd5558f4238859d9